content/intro-to-storybook/react/zh-TW/simple-component.md

```diff
@@ -41,95 +36,127 @@ export default function Task({ task: { id, title, state }, onArchiveTask, onPinT
 
-上面,我們基於 Todos 應用程式現有 HTML 結構 為 `Task`提供簡單的 markup .
+在上面，我們根據待辦事項 app 現有的 HTML 結構，把`工作事項`語法直覺地 render 出來。
 
-下面, 我們在 故事檔案中 構建 Task 的 三個測試狀態:
+接下來，要在 story 檔案蓋出工作事項的 3 種測試狀態：
 
-```javascript
+```js:title=src/components/Task.stories.js
 import React from 'react';
-import { storiesOf } from '@storybook/react';
-import { action } from '@storybook/addon-actions';
 
 import Task from './Task';
 
-export const task = {
-  id: '1',
-  title: 'Test Task',
-  state: 'TASK_INBOX',
-  updatedAt: new Date(2018, 0, 1, 9, 0),
+export default {
+  component: Task,
+  title: 'Task',
 };
 
-export const actions = {
-  onPinTask: action('onPinTask'),
-  onArchiveTask: action('onArchiveTask'),
-};
+const Template = args => <Task {...args} />;
 
-storiesOf('Task', module)
-  .add('default', () => <Task task={task} {...actions} />)
-  .add('pinned', () => <Task task={{ ...task, state: 'TASK_PINNED' }} {...actions} />)
-  .add('archived', () => <Task task={{ ...task, state: 'TASK_ARCHIVED' }} {...actions} />);
-```
+export const Default = Template.bind({});
+Default.args = {
+  task: {
+    id: '1',
+    title: 'Test Task',
+    state: 'TASK_INBOX',
+    updatedAt: new Date(2018, 0, 1, 9, 0),
+  },
+};
 
-Storybook 中有兩個基本的組織級別.
+export const Pinned = Template.bind({});
+Pinned.args = {
+  task: {
+    ...Default.args.task,
+    state: 'TASK_PINNED',
+  },
+};
 
-該元件 及 其 child 故事.
+export const Archived = Template.bind({});
+Archived.args = {
+  task: {
+    ...Default.args.task,
+    state: 'TASK_ARCHIVED',
+  },
+};
+```
 
-將每個故事 視為元件的排列. 您可以根據需要為每個元件 建立 儘可能多的故事.
+在 Storybook，組織方式有兩個基本階層：元件和其子 story。把每個 story 想成是該元件的排列 (permutation)。只要有必要，每個元件可以有無限多個 story。
 
 - **元件**
-  - 故事
-  - 故事
-  - 故事
+  - Story
+  - Story
+  - Story
+
+為了要讓 Storybook 知道正在撰寫文件的元件，先做好 `default` export，包含：
 
-要開始 Storybook,我們先運`行 註冊元件的`storiesOf()`函式. 我們為元件新增 _顯示名稱 - Storybook 應用程式側欄上顯示的名稱_.
+- `component` -- 元件它本人；
+- `title` -- 在 Storybook app 側邊欄找到元件的名字；
+- `excludeStories` -- story 檔案裡的 export，但不應被 Storybook render 為 story；
+- `argTypes` -- 設定每個 story 的[參數](https://storybook.js.org/docs/react/api/argtypes)行為。
 
-`action()`允許我們建立一個回撥, 當在 Storybook UI 的面板中 單擊這個 **action** 時 回撥觸發. 因此,當我們構建一個 pin 按鈕 時,我們將能夠在 測試 UI 中 確定按鈕單擊 是否成功.
+若要定義出各式各樣的 story，就為每個要測試的狀態 export 函式，以此產生 story。Story 就是根據設定好的狀態，回傳 render 出元素的函式（也就是有一組 prop 的元件），這跟 [Stateless Functional Component](https://reactjs.org/docs/components-and-props.html) 一樣。
 
-由於我們需要將 相同的一組操作 傳遞給 元件的所有排列,因此將它們捆綁到`actions`變數 並 使用 React`{...actions}`的`porps`擴充套件以立即傳遞它們. `<Task {...actions}>`相當於`<Task onPinTask={actions.onPinTask} onArchiveTask={actions.onArchiveTask}>`.
+既然已經排列出元件，賦值給稱為 `Template` 的變數會很方便。採用這樣的模式來做 story 可以減少撰寫和維護程式碼的份量。
 
-關於捆綁`actions`的另一個好處就是,你可以`export-暴露`它們,用於重用該元件的元件,我們稍後會看到.
+<div class="aside">
+💡 <code>Template.bind({})</code> 是 <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind">標準的 JavaScript</a> 技巧，用來複製函式。此技巧是用來讓每個 export 出來的 story 可以設定各自的 property，但使用完全相同的方法。
+</div>
```

Comment:
@ymcheung Looking at this aside, you can translate it at will. Minus the "story", which in this case relates to Storybook's stories, the rest is ok.